web/src/reducers/navigation.js

```diff
@@ -0,0 +1,17 @@
+import { NAVIGATION_SCROLL_TO_WAYPOINT } from '../actions/navigation';
+
+const initialState = {
+  activeSection: null
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case NAVIGATION_SCROLL_TO_WAYPOINT:
+      return { ...state, activeSection: action.data };
+    default:
+      return state;
+  }
+};
+
+export const selectActiveSection = ({ navigation: { activeSection } }) =>
+  activeSection;
```

Comment:
Thinking about whether reducers and selectors should be in the same files or separate files. I don't have strong feelings about it...

Reply:
They feel similar enough to me for it to not feel weird.